pdr_backend/pdr_dashboard/dash_components/plots.py

```diff
@@ -1,33 +1,81 @@
 from itertools import product
-from typing import List, Optional, Tuple, Union
+from typing import List, Optional, Tuple, Union, NamedTuple
 
 import plotly.graph_objects as go
 from enforce_typing import enforce_types
+from statsmodels.stats.proportion import proportion_confint
 
 from pdr_backend.cli.arg_feeds import ArgFeeds
 from pdr_backend.util.time_types import UnixTimeS
 
 
+class FiguresAndMetricsResult(NamedTuple):
+    accuracy_figure: go.Figure
+    profit_figure: go.Figure
+    costs_figure: go.Figure
+    avg_accuracy: Optional[float]
+    total_profit: Optional[float]
+    avg_stake: Optional[float]
+
+
+class AccInterval(NamedTuple):
+    acc_l: float
+    acc_u: float
+
+
+class ProcessedPayouts(NamedTuple):
+    slot_in_date_format: List[str]
+    accuracies: List[float]
+    profits: List[float]
+    stakes: List[float]
+    correct_predictions: int
+    predictions: int
+    acc_intervals: List[AccInterval]
+
+
 @enforce_types
-def process_payouts(payouts: List[dict]) -> tuple:
+def process_payouts(
+    payouts: List[dict], calculate_confint: bool = False
+) -> ProcessedPayouts:
     """
     Process payouts data for a given predictor and feed.
     Args:
         payouts (list): List of payouts data.
         predictor (str): Predictor address.
         feed (str): Feed contract address.
     Returns:
-        tuple: Tuple of slots, accuracies, profits, and stakes.
+        tuple: Tuple of slots, accuracies, profits, stakes.
     """
-    slots, accuracies, profits, stakes = [], [], [], []
-    profit = predictions = correct_predictions = 0
+    (
+        _,
+        accuracies,
+        profits,
+        stakes,
+        correct_predictions,
+        predictions,
+        acc_intervals,
+    ) = ProcessedPayouts([], [], [], [], 0, 0, [])
```

Comment:
I mean you can initialize `processed_payouts = ProcessedPayouts([], [], [], [], 0, 0, [])` without unpacking the tuple.